sklearn/metrics/cluster/_unsupervised.py

```diff
@@ -464,3 +470,196 @@ def davies_bouldin_score(X, labels):
+    labels,
+    metric="euclidean",
+    d=None,
+    per_cluster_scores=False,
+    strict_cluster_size_validation=False,
+    mst_raw_dist=False,
+    verbose=False,
+    **kwd_args,
+):
+    """
+    Compute the density based cluster validity index for a clustering.
+
+    Contrary to alternative, more widely used unsupervised clustering
+    metrics, it does not implicitly assume clusters of spherical shape.
+    This is accomplished by circumventing the need for calculating
+    cluster centroids as part of its mathematical definition. Instead,
+    it evaluates the quality of clusters based on the concepts of
+    density separation (highest density area between clusters) and
+    density sparseness (lowest density area within a cluster).
+    These concepts are propped up on the underlying construct of
+    minimum spanning trees, much like the HDBSCAN clustering algorithm.
+    The metric also accounts for noise points as part of its definition.
+    This implementation assumes such points to be labeled with -1.
+
+    The best value that can be returned is 1 and the worst value is -1.
+
+    Parameters
+    ----------
+    X : array (n_samples, n_features) or (n_samples, n_samples)
+        The input data of the clustering. This can be the data, or, if
+        metric is set to `precomputed` the pairwise distance matrix used
+        for the clustering.
+
+    labels : array (n_samples)
+        The label array output by the clustering, providing an integral
+        cluster label to each data point, with -1 for noise points.
+
+    metric : optional, str (default 'euclidean')
+        The metric used to compute distances for the clustering (and
+        to be re-used in computing distances for mr distance). If
+        set to `precomputed` then X is assumed to be the precomputed
+        distance matrix between samples.
+
+    d : optional, int (or None) (default None)
+        The number of features (dimension) of the dataset. This need only
+        be set in the case of metric being set to `precomputed`, where
+        the ambient dimension of the data is unknown to the function.
+
+    per_cluster_scores : optional, bool (default False)
+        Whether to return the validity index for individual clusters.
+        Defaults to False with the function returning a single float
+        value for the whole clustering.
+
+    strict_cluster_size_validation : optional, bool (default False)
+        If True, fail fast on cluster sizes for which DBCV is undefined.
+
+    mst_raw_dist : optional, bool (default False)
+        If True, the MST's are constructed solely via 'raw' distances
+        (depending on the given metric, e.g. euclidean distances)
+        instead of using mutual reachability distances.
+        Thus setting this parameter to True avoids using 'all-points-core-distances'.
+        This is advantageous specifically in the case of elongated clusters
+        that lie in close proximity to each other <citation needed>.
```

Comment:
**TLDR: there is no paper we can cite here. I'll just remove this part of the docstring if you want.**

**practical considerations:**
I'd argue the actual functionality of `mst_raw_dist` should remain either way: it's not like it adds some unproven code, it simply turns off the use of mutual reachability distances, instead directly using the raw, underlying distances to construct the MSTs.

- in some cases, limited compute resources might dictate falling back on simple, non-composite distances when dealing with extra large datasets anyway
- I have data (to be precise: not simply data itself, but instead a setup for a reproducible [experiment that, when ran, produces such data](https://github.com/luis261/clustering-validity-comparison)) which systematically demonstrates that constructing the MST's with simple, non-MRD distances can actually be beneficial irrespective of runtime performance, since it additionally positively affects the performance of the metric itself (correlation with ground-truth as per DOI:10.1002/sam.10080)

I just wanted to be transparent here, if sklearn policy strictly dictates that such "experimental" functionality absolutely can't remain without being backed by published literature, I understand, no hard feelings.

<details>
<summary>the whole story regarding the missing citation (theoretical background/implications)</summary>
Alright, so bear with me, here's some context: while running the benchmark(s) of clustering validity metrics that backed the quantitative part of my thesis I noticed that, compared to traditional, centroid based metrics, DBCV didn't perform as well as you'd expect it to. It actually performed just as bad as the traditional metrics "where it matters" (that is: when evaluated against the same, density based synthetic datasets which were in fact specifically engineered to allow the score to pronounce the non-spherical nature in which it's supposed to operate/evaluate).

This unexpected result prompted me to start investigating. I ultimately tracked down the culprit, which is that the composite distances that are used to construct the MSTs were actually surpassing the basic, underlying distances (e.g. euclidean) by an order of magnitude in some cases (the max I ever measured was an 18 fold increase). This inflation of the core distances leads to comparatively long distances in the constructed minimum spanning trees, in turn resulting in a relatively high density sparseness, which leads to a worse score overall.

<details>
<summary>peeling back one more layer, into the mathematical weeds of the score</summary>
It may look like it at first glance, but I don't think that what I measured with the up to 18 fold increase actually constitutes a disproof by counterexample regarding the following assertions of the DBCV paper:

"[...] the core distance of an object should be approximately to the distance of a Kth nearest neighbor where K is not too large (representing a small neighborhood of the object)" (at least when going with a rather generous definition of "too large" or "small neighborhood")

The paper also states previously that:
"since in the definition of a mutual reachability distance the core distance of an object is compared to the distances of the object to other objects in the cluster, the core distance should be comparable to these distances",
however "these distances" refers to all (pairwise) distances of that point to any other points in the cluster (not just those in a "small neighborhood"), which also includes the distance to the most distant point at the opposite end of the cluster. For elongated clusters, this diameter can be comparatively large.

There seems to be a tradeoff though between avoiding random noise points from shortening what should otherwise be a sparse link in an MST by acting as a "bridge" while also not having the overall result of the metric succumb to the bias against elongated shapes that implicitly resides in composite distances that incorporate KNNs (with large/maximum K).

Besides entirely cutting the MRDs out of the scoring procedure or switching to squared euclidean distances as the authors suggest, I'd recommend exploring using core point distances in local neighborhoods instead of all points core distances. It would make sense to parameterize this `min_samples` value to allow users to e.g. increase it for particularly large/dense datasets. By only considering the local neighborhoods, one can circumvent the bias against elongated clusters.

Either way, my benchmark has demonstrated that the descriptions of some of the supposed properties of DBCV turn out to be rather "generous" when compared with the performance/output of its default configuration. While not strictly incorrect, they turn out to be of diminished significance/bearing with regards to practical applications (regarding the MRD components only).
</details>

Replacing the MRDs with raw distances remedied the problem, resulting in a strongly (positive) correlation of the score and the similarity of the labels assigned by the clustering with the labelling according to the underlying ground truth (while still retaining equivalent performance when operating on spherical datasets).

I had originally planned to pursue a paper which would build on the aforementioned benchmark by expanding on it in higher dimensions with more diverse datasets (my benchmark was limited to 2d, with an underlying normal distribution, pre elongation transform). I speculate that the curse of dimensionality would even impact the MRDs more than it does the raw distances. On that note, it might also be interesting to try out alternative distance metrics besides just euclidean in the benchmarks.

Besides just calling the benefit of MRDs into question though, the paper might also explore if they have some benefit after all, since the DBCV paper states that the "higher weight of the contribution of closer objects" could be amplified "by using the squared Euclidean distance instead of Euclidean distance as dissimilarity" which could potentially constitute an alternative for solving the inflation problem without getting rid of MRDs entirely (see detailed section above).

<details>
<summary>
as an aside: HDBSCAN remains unaffected either way
</summary>
even if confirmed, there are no implications for HDBSCAN, despite it being propped up on MSTs of composite distances, much like DBCV. HDBSCAN doesn't work with "all points core distances", instead opting for core distances that only include the local neighborhood up until a limit in the calculation (sklearn implementation defaults to `min_samples` of 5: https://github.com/scikit-learn/scikit-learn/blame/main/sklearn/cluster/_hdbscan/hdbscan.py).

https://hdbscan.readthedocs.io/en/latest/how_hdbscan_works.html
</details>

Ideally, I'd collaborate with the original DBCV authors, thus inviting additional oversight/verification. They would already be familiar with the methodology, considering there is an author overlap between DOI:10.1002/sam.10080 and the DBCV/HDBSCAN papers respectively.

I haven't even gotten around to proposing it though and I'm not sure if they'd really be interested. It's probably not realistic I guess. I have taken on a non-research, non-datasci job after graduating anyway and neither have the time nor energy to see this through currently. I don't see myself pursuing/going back to academia in the future either, so that path is pretty much a dead end.

Sorry about the delusional essay, but in case you read this, now you know the deal about the missing citation.
</details>